test/src/main/java/org/apache/accumulo/test/start/KeywordStartIT.java

```diff
@@ -201,6 +203,8 @@ public void checkHasMain() {
     expectSet.add(Shell.class);
     expectSet.add(SimpleGarbageCollector.class);
     expectSet.add(TabletServer.class);
+    expectSet.add(ZooInfoViewer.class);
+    expectSet.add(ZooPropSetTool.class);
```

Comment:
What determines when a tool should have a main method?  The javadoc in `KeywordExecutable` states how, if it is desired to have a "redundant main method" what it should look like.  Certainly using `accumulo tool-name` is convenient, but I do not know when someone might prefer using a main method, or which tools "require" a main method, other than the list in KeywordStartIT, which I think over time may be testing tools that "have a main method" vs. test tools that actually "require" a main main, but have one for convenience (or in my case added because of ignorance of the difference.)

The Keyword and main support seems to mainly be boiler-plate between utilities and I was trying to maintain similar functionality between various tools so that would provide similar experiences rather than the user needing know tool-by-tool which options it would support,

Reply:
I removed main in d11c27ea95 and I think the KeywordStartIT was updated correctly - the test passes, but unclear if this is how the test should be used for non-main tools.

Reply:
Tools typically don't have a main method. Everything generally gets launched through our Start entry point. However, many existing tools happened to have a main method. So, the test was an attempt to make sure we didn't accidentally remove a main method that users might be using in scripts.

Other than Start, and tools that don't have a keyword, we don't have a requirement for any tools to have a "main" method at all. I think you updated the IT correctly.

This is out of scope for this PR, but we may want to add a test to check that we don't introduce new main classes unintentionally. Currently we only have a test that checks that we're preserving the existing ones... but a better test would be to check through all our classes, and check each one for having a main method, and failing if it finds any we haven't identified as needing one.